### DISCUSSION TITLE
Loosing Control checkbox value after rerendering when using custom parse and format methods for field's value processing

### DISCUSSION THREAD

**Discussion by sergeyivanovv89** [react-hook-form/react-hook-form#5949] [Q&A] [answered]
**Describe the bug**
We have a simple two-step form and Control with reactstrap checkbox. And if we apply our custom parse and format methods for the field we lose its value after rerendering.

**To Reproduce**

1. Click on 'Next' button (the checkbox is checked initially)
2. Click on 'Prev' button
3. See error

**Codesandbox link**
https://codesandbox.io/s/quirky-haibt-pmgos

**Expected behavior**
The checkbox must be checked but it doesn't.

**Desktop (please complete the following information):**

- OS: Win7
- Browser chrome
- Version 91.0.4472.124 (x64)

**Comment by bluebill1049** [comment, marked as answer]
<img width="997" alt="Screen Shot 2021-07-15 at 9 38 02 am" src="https://user-images.githubusercontent.com/10513364/125706102-ce4353a3-3df0-4ae9-a152-677502a60b09.png">

the data gets transformed and it's no longer the original shape.
